core/src/test/java/dev/streamx/cli/ingestion/BaseIngestionCommandTest.java

```diff
@@ -0,0 +1,44 @@
+package dev.streamx.cli.ingestion;
+
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static dev.streamx.clients.ingestion.StreamxClient.PUBLICATIONS_ENDPOINT_PATH_V1;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.github.tomakehurst.wiremock.junit5.WireMockExtension;
+import io.quarkus.test.junit.main.LaunchResult;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public abstract class BaseIngestionCommandTest {
+
+  @RegisterExtension
+  protected static WireMockExtension wm = WireMockExtension.newInstance()
+      .options(wireMockConfig().dynamicPort())
+      .configureStaticDsl(true)
+      .build();
+
+  @BeforeEach
+  void setup() {
+    initializeWiremock();
+  }
+
+  protected abstract void initializeWiremock();
+
+  protected static String getIngestionUrl() {
+    return "http://localhost:" + wm.getPort();
+  }
+
+  protected static String getPublicationPath(String channel, String key) {
+    return PUBLICATIONS_ENDPOINT_PATH_V1 + "/" + channel + "/" + key;
+  }
+
+  protected static void expectSuccess(LaunchResult result) {
+    assertThat(result.exitCode()).isZero();
+    assertThat(result.getErrorOutput()).isEmpty();
+  }
+
+  protected static void expectError(LaunchResult result, String expectedErrorOutput) {
+    assertThat(result.exitCode()).isNotZero();
+    assertThat(result.getErrorOutput()).isEqualTo(expectedErrorOutput);
```

Comment:
Seal two assertions into one method: if error message is not empty - then exit code must not be zero (and vice versa)

Reply:
Second thing is: some tests verified only parts of error messages (they used contains and not isEqualTo). That's why we didn't see that actually the error messages displayed the user's command line were very long sometimes (this observation was the reason why @michalcukierman created this task)